Author: Luca Boccassi <bluca@debian.org>
Description: LoadCredentials: do not assert on invalid syntax
 LoadCredentials=foo causes an assertion to be triggered, as we
 are not checking that the rvalue's right hand side part is non-empty
 before using it in unit_full_printf.
 .
 NOTE: it will still apply cleanly, but drop from v249 as it's fixed by a new
 feature
Forwarded: yes, https://github.com/systemd/systemd-stable/pull/99
--- a/src/core/load-fragment.c
+++ b/src/core/load-fragment.c
@@ -4569,7 +4569,7 @@
         r = extract_first_word(&p, &word, ":", EXTRACT_DONT_COALESCE_SEPARATORS);
         if (r == -ENOMEM)
                 return log_oom();
-        if (r <= 0) {
+        if (r <= 0 || isempty(p)) {
                 log_syntax(unit, LOG_WARNING, filename, line, r, "Invalid syntax, ignoring: %s", rvalue);
                 return 0;
         }
